reconcile/utils/terrascript_client.py

```diff
@@ -3374,7 +3378,33 @@ def populate_tf_resource_s3_cloudfront_public_key(self, resource,
         for tf_resource in tf_resources:
             self.add_resource(account, tf_resource)
 
-    def populate_tf_resource_alb(self, resource, namespace_info):
+    def _get_alb_target_ips_by_openshift_service(self,
+                                                 openshift_service,
+                                                 account_name,
+                                                 namespace_info,
+                                                 ocm_map):
+        account = self.accounts[account_name]
+        awsapi = AWSApi(1, [account],
+                        settings=self.settings,
+                        init_users=False)
+        cluster = namespace_info['cluster']
+        ocm = ocm_map.get(cluster['name'])
+        account['assume_role'] = \
+            ocm.get_aws_infrastructure_access_terraform_assume_role(
+                cluster['name'],
+                account['uid'],
+                account['terraformUsername'],
+            )
+        account['assume_region'] = cluster['spec']['region']
+        service_name = \
+            f"{namespace_info['name']}/{openshift_service}"
+        return awsapi.get_alb_network_interface_ips(
```

Comment:
Should we add a safeguard to verify that `get_alb_network_interface_ips` did not return an empty set? Services should always have at least 1 IP address (single-az) and I'm worried that if one day the tags change or maybe if the AWS API call is rate limited or something else that doesn't raise an exception we could end up removing all IP addresses from the ALB

Reply:
handled in https://github.com/app-sre/qontract-reconcile/pull/2008/commits/b0e9d09383fe9f79c0488e3186840bef984de6e9